src/fe-common/irc/fe-irc-channels.c

```diff
@@ -31,6 +31,51 @@
 #include "fe-windows.h"
 #include "window-items.h"
 
+int fe_channel_is_opchannel(IRC_SERVER_REC *server, const char *target)
+{
+	const char *statusmsg;
+
+	/* Quick check */
+	if (server == NULL || server->prefix[(int)(unsigned char)*target] == 0)
+		return FALSE;
+
+	statusmsg = g_hash_table_lookup(server->isupport, "statusmsg");
+	if (statusmsg == NULL)
+		statusmsg = "@+";
+
+	return strchr(statusmsg, *target) != NULL;
+}
+
+const char *fe_channel_skip_prefix(IRC_SERVER_REC *server, const char *target)
+{
+	const char *statusmsg;
+
+	/* Quick check */
+	if (server == NULL || server->prefix[(int)(unsigned char)*target] == 0)
```

Comment:
The cast's here deserves a comment in the code. Make a follow up patch which cleans that up a bit please :-)